src/relentless/optimize/method.py

```diff
@@ -54,6 +56,21 @@ def stop(self, value):
             raise TypeError("The stopping criterion must be a ConvergenceTest.")
         self._stop = value
 
+    def setup_directory(self, directory=None, overwrite=False):
+        if directory is not None:
+            directory = data.Directory.cast(directory, create=mpi.world.rank_is_root)
+            if not directory.is_empty():
+                if overwrite is True:
+                    if mpi.world.rank_is_root:
+                        directory.clear_contents()
+                else:
+                    raise OSError(
+                        "Directory {} is not empty and overwrite is not True".format(
+                            directory.path
+                        )
+                    )
+            mpi.world.barrier()
```

Comment:
I think the error might be because you are trying to check the directory contents on other ranks before it is created. By moving the barrier up, we can ensure it exists before calling the other operations. Then, we clear the contents only the root rank and put up another barrier if we need to clear.
```suggestion
        directory = data.Directory.cast(directory, create=mpi.world.rank_is_root)
        mpi.world.barrier()
        if not directory.is_empty():
            if overwrite is True:
                if mpi.world.rank_is_root:
                    directory.clear_contents()
                mpi.world.barrier()
            else:
                raise OSError(
                    "Directory {} is not empty and overwrite is not True".format(
                        directory.path
                    )
                )
        
```